airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py

```diff
@@ -28,8 +28,34 @@ class CartesianProductStreamSlicer(StreamSlicer):
     """
 
     def __init__(self, stream_slicers: List[StreamSlicer]):
+        """
+        :param stream_slicers: Underlying stream slicers
+        """
         self._stream_slicers = stream_slicers
 
+    def update_cursor(self, stream_slice: Mapping[str, Any], last_record: Optional[Mapping[str, Any]] = None):
+        for slicer in self._stream_slicers:
+            slicer.update_cursor(stream_slice, last_record)
+
+    def request_params(self) -> Mapping[str, Any]:
+        return dict(ChainMap(*[s.request_params() for s in self._stream_slicers]))
+
+    def request_headers(self) -> Mapping[str, Any]:
+        return dict(ChainMap(*[s.request_headers() for s in self._stream_slicers]))
+
+    def request_body_data(self) -> Optional[Union[Mapping, str]]:
+        return dict(ChainMap(*[s.request_body_data() for s in self._stream_slicers]))
+
+    def request_body_json(self) -> Optional[Mapping]:
+        return dict(ChainMap(*[s.request_body_json() for s in self._stream_slicers]))
+
+    def get_stream_state(self) -> Optional[Mapping[str, Any]]:
+        states = list(filter(lambda s: s, [slicer.get_stream_state() for slicer in self._stream_slicers]))
```

Comment:
it was to remove `None` states, but this can be simplified by removing the `Optional` from the type hint and returning an empty `{}` instead of `None`